projects/plugins/jetpack/extensions/shared/components/product-management-controls/use-open-block-sidebar.js

```diff
@@ -1,22 +1,24 @@
 /**
  * WordPress dependencies
  */
-import { useDispatch, select } from '@wordpress/data';
+import { useDispatch } from '@wordpress/data';
+import {
+	getGutenbergContext,
+	POST_EDITOR_CONTEXT,
+	SITE_EDITOR_CONTEXT,
+	WIDGET_EDITOR_CONTEXT,
+} from '../gutenberg/context/resolver';
 
 export default function useOpenBlockSidebar() {
+	const gutenbergContext = getGutenbergContext();
 	const { enableComplementaryArea } = useDispatch( 'core/interface' );
-	const isSiteEditor = !! select( 'core/edit-site' );
-	const isWidgetsEditor = !! select( 'core/edit-widgets' );
 
-	return () => {
-		if ( isSiteEditor ) {
-			enableComplementaryArea( 'core/edit-site', 'edit-site/block-inspector' );
-			return;
-		}
-		if ( isWidgetsEditor ) {
-			enableComplementaryArea( 'core/edit-widgets', 'edit-widgets/block-inspector' );
-			return;
-		}
-		enableComplementaryArea( 'core/edit-post', 'edit-post/block' );
-	};
+	switch ( gutenbergContext ) {
+		case SITE_EDITOR_CONTEXT:
+			return () => enableComplementaryArea( 'core/edit-site', 'edit-site/block-inspector' );
+		case WIDGET_EDITOR_CONTEXT:
+			return () => enableComplementaryArea( 'core/edit-widgets', 'edit-widgets/block-inspector' );
+		case POST_EDITOR_CONTEXT:
+			return () => enableComplementaryArea( 'core/edit-post', 'edit-post/block' );
```

Comment:
This is formally correct, but I'd probably keep the post editor as the default return _just in case_.

Reply:
Defaulted to post editor.